lib/routes/nwc/widgets/nwc_connect_bottom_sheet.dart

```diff
@@ -0,0 +1,563 @@
+                            if (!_showBudgetFields) {
+                              return null;
+                            }
+                            if (trimmedValue.isEmpty && resetTimeValue.isNotEmpty) {
+                              return 'Enter max budget';
+                            }
+                            if (trimmedValue.isNotEmpty && int.tryParse(trimmedValue) == null) {
+                              return 'Please enter a valid number';
+                            }
+                            return null;
+                          },
+                        ),
+                        const Divider(),
+                        TextFormField(
+                          controller: _resetTimeController,
+                          keyboardType: TextInputType.number,
+                          decoration: InputDecoration(
+                            labelText: 'Reset Time (Optional)',
+                            hintText: 'Enter Time in seconds',
+                            border: const OutlineInputBorder(),
+                            errorBorder: OutlineInputBorder(
+                              borderSide: BorderSide(color: themeData.colorScheme.error),
+                            ),
+                            focusedErrorBorder: OutlineInputBorder(
+                              borderSide: BorderSide(color: themeData.colorScheme.error),
+                            ),
+                          ),
+                          validator: (String? value) {
+                            final String trimmedValue = value?.trim() ?? '';
+                            final String maxBudgetValue = _maxBudgetController.text.trim();
+                            final String expiryTimeValue = _expiryTimeController.text.trim();
+
+                            if (!_showBudgetFields) {
+                              return null;
+                            }
+                            if (trimmedValue.isEmpty && maxBudgetValue.isNotEmpty) {
+                              return 'Enter reset time';
+                            }
+                            if (trimmedValue.isNotEmpty && int.tryParse(trimmedValue) == null) {
+                              return 'Please enter a valid number';
+                            }
+
+                            if (trimmedValue.isNotEmpty && expiryTimeValue.isNotEmpty) {
+                              final int? resetTime = int.tryParse(trimmedValue);
+                              final int? expiryTime = int.tryParse(expiryTimeValue);
+                              if (resetTime != null && expiryTime != null && resetTime > expiryTime) {
+                                return 'Reset time cannot be greater than expiry time';
+                              }
+                            }
+
+                            return null;
+                          },
+                        ),
```

Comment:
Let's make this a dropdown menu, or a column of buttons titled 'Budget Renewal' with the following options:
`DAILY`, `WEEKLY`, `MONTHLY`, `YEARLY`, `NEVER`